### PR TITLE
Integration test updates for grub_dpkg

### PR DESCRIPTION
**Pull request by TheRealFalcon**
## Proposed Commit Message

Use individual commits

## Merge type

- [ ] Squash merge using "Proposed Commit Message"
- [x] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: tests/integration_tests/test_defaults.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        expected_inactive = {
+            "apt_pipelining",
+            "ansible",
+            "bootcmd",
+            "ca_certs",
+            "chef",
+            "disable_ec2_metadata",
+            "disk_setup",
+            "fan",
+            "grub_dpkg",
```
Comment (aciba90) [inline tests/integration_tests/test_defaults.py:35]: Won't this fail with dailies on `oracular-` after #5842 is fixed?

**Review by aciba90 — COMMENTED** [review]
Thanks for the reorganization of the tests!

`grub_dpkg` is disabled by default in upstream code and we are adding the check accordingly. But, we are going to patch that behavior out for stable releases, should we conditionally add `grub_dpkg` as inactive module in the test check depending on the Ubuntu release? Or is that something we will fix in another future PR?

File: tests/integration_tests/test_defaults.py (hunk 1 above)
Comment (TheRealFalcon) [reply tests/integration_tests/test_defaults.py]: Updated

File: tests/integration_tests/test_logging.py [hunk 2]
```diff
@@ -1,22 +0,0 @@
-"""Integration tests relating to cloud-init's logging."""
```
Comment (holmanb) [inline tests/integration_tests/test_logging.py:1]: This is my go-to "run a single test", but I'll be fine :P

File: tests/integration_tests/modules/test_ca_certs.py [hunk 3]
```diff
@@ -105,59 +101,7 @@ def test_cert_installed(self, class_client: IntegrationInstance):
         )
 
     def test_clean_log(self, class_client: IntegrationInstance):
-        """Verify no errors, no deprecations and correct inactive modules in
-        log.
-        """
-        log = class_client.read_from_file("/var/log/cloud-init.log")
+        """Verify no errors or deprecations in log."""
+        log = class_client.execute("cat /var/log/cloud-init-output.log")
```
Comment (holmanb) [inline tests/integration_tests/modules/test_ca_certs.py:105]: why prefer `.execute("cat ..")` over `.read_from_file()`?

File: .github/workflows/integration.yml [hunk 4]
```diff
@@ -68,4 +69,4 @@ jobs:
           echo "[lxd]" > /home/$USER/.config/pycloudlib.toml
       - name: Run integration Tests
         run: |
-          CLOUD_INIT_CLOUD_INIT_SOURCE="$(ls ${{ runner.temp }}/cloud-init*.deb)" CLOUD_INIT_OS_IMAGE=oracular tox -e integration-tests-ci -- --color=yes tests/integration_tests/
+          CLOUD_INIT_CLOUD_INIT_SOURCE="$(ls ${{ runner.temp }}/cloud-init*.deb)" CLOUD_INIT_OS_IMAGE=plucky tox -e integration-tests-ci -- --color=yes tests/integration_tests/
```
Comment (holmanb) [inline .github/workflows/integration.yml:72]: We could also just reference `env.RELEASE` here - right? +1, either way

File: tests/integration_tests/test_defaults.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
+"""Tests here shouldn't require any sort of user data or instance setup."""
```
Comment (holmanb) [inline tests/integration_tests/test_defaults.py:1]: I like this idea a lot. Common case tests like this have many benefits:

- fast (class_client)
- high-impact (add coverage that benefits most users)
- relatively easy to read/write (not covering some bespoke feature/cloud)

**Review by holmanb — APPROVED** [review]
Thanks for this @TheRealFalcon! Looks good to me!

File: tests/integration_tests/modules/test_ca_certs.py (hunk 3 above)
Comment (TheRealFalcon) [reply tests/integration_tests/modules/test_ca_certs.py:105]: I...have no idea why I did that :joy:

**Comment by TheRealFalcon** [comment]
@holmanb , updated based on comments

File: .github/workflows/integration.yml [hunk 6]
```diff
@@ -17,7 +17,8 @@ defaults:
     shell: sh -ex {0}
 
 env:
-  RELEASE: oracular
+  # If you change this, don't forget to also update the line that invokes tox
```
Comment (holmanb) [inline .github/workflows/integration.yml:20]: I guess this line isn't required anymore then?

**Review by holmanb — APPROVED** [review]
One nitpick, but otherwise looks good!

**Pull request merged**